src/NUnitFramework/framework/Constraints/UniqueItemsConstraint.cs

```diff
@@ -60,5 +69,123 @@ protected override bool Matches(IEnumerable actual)
 
             return true;
         }
+
+        private bool? TryFastAlgorithm(IEnumerable actual)
+        {
+            // If the user specified any external comparer with Using, exit
+            if (UsingExternalComparer)
+                return null;
```

Comment:
Just a note: if the external comparer is an `IEqualityComparer<>`, we can use it with the `ItemsUnique` method below and be super fast.

Reply:
Correct. That could be another step in improving this, but it would involve analyzing all the external comparers to see if any of them applied and that effort would be wasted if we end up rewriting the NUnitEqualityComparer anyway. My goal in this PR was to do as much as I could while remaining inside the UniqueItemsConstraint. 

Since the issue is about `CollectionAssert`, we can never have either an external comparer or a case-insensitive comparison. You have to use `Is.Unique` to specify those.